front/lib/swr/data_source_views.ts

```diff
@@ -143,18 +145,27 @@ export function useDataSourceViewContentNodes({
     }); // Serialize with body to ensure uniqueness.
   }, [url, body]);
 
-  const { data, error, mutate } = useSWRWithDefaults(fetchKey, async () => {
-    if (!url) {
-      return undefined;
-    }
-
-    return postFetcher([url, { internalIds, parentId, viewType }]);
-  });
+  const { data, error, mutate, mutateRegardlessOfQueryParams } =
+    useSWRWithDefaults(
+      fetchKey,
+      async () => {
+        if (!url) {
+          return undefined;
+        }
+
+        return postFetcher([url, { internalIds, parentId, viewType }]);
+      },
+      {
+        disabled: disabled || !viewType,
+      }
+    );
 
   return {
     isNodesError: !!error,
     isNodesLoading: !error && !data,
     mutateDataSourceViewContentNodes: mutate,
+    mutateDataSourceViewContentNodesRegardlessOfQueryParams:
+      mutateRegardlessOfQueryParams,
```

Comment:
nit: when i have the opportunity, i refactor to keep `mutate` and `mutateRegardlessOfQueryParams` in the hook. I rename at usage in components instead (_if necessary_ to avoid confusion).

Reply:
Agree it's better 👍  doing